rwlock Libos Regression test fails on Ubuntu 22.04, Almalinux, Rockylinux, CentOS 9

### Description of the problem

rwlock Libos Regression test fails on Ubuntu 22.04, Almalinux, Rockylinux, CentOS 9. The test is just stuck.

### Steps to reproduce

1. Build Gramine
2. cd libos/test/regression
3. gramine-test build (Build Libos Regression Test)
4. python3 -m pytest -s -v -k test_001_rwlock (Run test)

### Expected results

Test OK

### Actual results

```
python3 -m pytest -s -v -k test_001_rwlock
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.10.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/intel/anjali/gramine, configfile: pytest.ini
plugins: xdist-2.5.0, forked-1.4.0
collected 151 items / 150 deselected / 1 selected                                                                                                                          

test_libos.py::TC_00_Unittests::test_001_rwlock FAILED

================================================================================= FAILURES =================================================================================
_____________________________________________________________________ TC_00_Unittests.test_001_rwlock ______________________________________________________________________
test_libos.py:40: in test_001_rwlock
    stdout, _ = self.run_binary(['rwlock', str(instances), str(iterations), str(readers_num),
/home/intel/gramine_install/usr/lib/python3.10/site-packages/graminelibos/regression.py:224: in run_binary
    _returncode, stdout, stderr = run_command(cmd, timeout=timeout, **kwds)
/home/intel/gramine_install/usr/lib/python3.10/site-packages/graminelibos/regression.py:157: in run_command
    raise AssertionError('Command {} timed out after {} s'.format(cmd, timeout))
E   AssertionError: Command ['/home/intel/gramine_install/usr/lib/x86_64-linux-gnu/gramine/direct/loader', '/home/intel/gramine_install/usr/lib/x86_64-linux-gnu/gramine/direct/libpal.so', 'init', 'rwlock', '5', '100', '10', '3', '100'] timed out after 45 s
========================================================================= short test summary info ==========================================================================
FAILED test_libos.py::TC_00_Unittests::test_001_rwlock - AssertionError: Command ['/home/intel/gramine_install/usr/lib/x86_64-linux-gnu/gramine/direct/loader', '/home/in...
==================================================================== 1 failed, 150 deselected in 45.54s ====================================================================
```

### Gramine commit hash

9ead920f7c44455b0cef3c27dcd496d0a908bfec

--- a/libos/include/arch/x86_64/gramine_entry_api.h
+++ b/libos/include/arch/x86_64/gramine_entry_api.h
@@ -43,6 +43,20 @@ __asm__(
     ".endm\n"
 );
 
+/* Magic syscall number for Gramine custom calls */
+#define GRAMINE_CUSTOM_SYSCALL_NR 0x100000
+
+/* Some GCC versions warn about unused parameters in naked functions. Bug? */
+__attribute__((naked)) static int gramine_call(int number __attribute__((unused)),
+                                               unsigned long arg1 __attribute__((unused)),
+                                               unsigned long arg2 __attribute__((unused))) {
+    __asm__ (
+        "mov $" GRAMINE_XSTR(GRAMINE_CUSTOM_SYSCALL_NR) ", %eax\n"
+        "GRAMINE_SYSCALL\n"
+        "ret\n"
+    );
+}
+
 #undef GRAMINE_XSTR
 #undef GRAMINE_STR
 
@@ -60,19 +74,6 @@ enum {
     GRAMINE_CALL_RWLOCK_WRITE_LOCK,
     GRAMINE_CALL_RWLOCK_WRITE_UNLOCK,
 };
-
-/* Magic syscall number for Gramine custom calls */
-#define GRAMINE_CUSTOM_SYSCALL_NR 0x100000
-
-static inline int gramine_call(int number, unsigned long arg1, unsigned long arg2) {
-    int ret;
-    __asm__ volatile(
-        "GRAMINE_SYSCALL"
-        : "=a"(ret)
-        : "0"(GRAMINE_CUSTOM_SYSCALL_NR), "D"(number), "S"(arg1), "d"(arg2)
-        : "rcx", "r11", "memory");
-    return ret;
-}
 
 static inline int gramine_register_library(const char* name, unsigned long load_address) {
     return gramine_call(GRAMINE_CALL_REGISTER_LIBRARY, (unsigned long)name, load_address);

